src/profile-logic/import/dhat.js

```diff
@@ -290,8 +294,22 @@ export function attemptToConvertDhat(json: mixed): Profile | null {
         stackTable.category.push(otherCategory);
         stackTable.category.push(otherSubCategory);
         stackTable.prefix.push(prefix);
+
+        const candidateList = postfix.get(prefix);
+        if (candidateList) {
+          // Append us to the list of possible stack indexes of our parent.
+          candidateList.push(stackIndex);
+        } else {
+          // We are the first descendents of our parent.
+          postfix.set(prefix, [stackIndex]);
+        }
+
         // The stack index already points to this spot.
         stackTable.length++;
+        // Since we just created a stack index, the next frames necessarily don't have an existing stack index.
+        candidateStackTables = [];
+      } else {
+        candidateStackTables = postfix.get(stackIndex);
```

Comment:
I think you can run `candidateStackTables = postfix.get(prefix)` at the start of the loop (right before `if (candidateStackTables)`), and get rid of this line as well as the previous line and the line `let candidateStackTables = postfix.get(null);` before the loop.